aws/service.ts

```diff
@@ -220,51 +220,45 @@ function computeImage(
         imageName: string,
         container: cloud.Container,
-        ports: ExposedPorts | undefined,
+        exposedPortOpt: ExposedPort | undefined,
         repository: aws.ecr.Repository | undefined): pulumi.Output<ImageOptions> {
 
     // Start with a copy from the container specification.
     const preEnv: Record<string, pulumi.Input<string>> =
         Object.assign({}, container.environment || {});
 
     // Now add entries for service discovery amongst containers exposing endpoints.
-    if (ports) {
-        for (const service of Object.keys(ports)) {
-            let firstPort = true;
-            const serviceEnv = makeServiceEnvName(service);
-            for (const port of Object.keys(ports[service])) {
-                const info = ports[service][parseInt(port, 10)];
-                const hostname = info.host.dnsName;
-                const hostport = info.hostPort.toString();
-                const hostproto = info.hostProtocol;
-                // Populate Kubernetes and Docker links compatible environment variables.  These take the form:
-                //
-                //     Kubernetes:
-                //         {SVCNAME}_SERVICE_HOST=10.0.0.11 (or DNS name)
-                //         {SVCNAME}_SERVICE_PORT=6379
-                //     Docker links:
-                //         {SVCNAME}_PORT=tcp://10.0.0.11:6379 (or DNS address)
-                //         {SVCNAME}_PORT_6379_TCP=tcp://10.0.0.11:6379 (or DNS address)
-                //         {SVCNAME}_PORT_6379_TCP_PROTO=tcp
-                //         {SVCNAME}_PORT_6379_TCP_PORT=6379
-                //         {SVCNAME}_PORT_6379_TCP_ADDR=10.0.0.11 (or DNS name)
-                //
-                // See https://kubernetes.io/docs/concepts/services-networking/service/#discovering-services and
-                // https://docs.docker.com/engine/userguide/networking/default_network/dockerlinks/ for more info.
-                if (firstPort) {
-                    preEnv[`${serviceEnv}_SERVICE_HOST`] = hostname;
-                    preEnv[`${serviceEnv}_SERVICE_PORT`] = hostport;
-                }
-                firstPort = false;
-
-                const fullHost = hostname.apply(h => `${hostproto}://${h}:${hostport}`);
-                preEnv[`${serviceEnv}_PORT`] = fullHost;
-                preEnv[`${serviceEnv}_PORT_${port}_TCP`] = fullHost;
-                preEnv[`${serviceEnv}_PORT_${port}_TCP_PROTO`]= hostproto;
-                preEnv[`${serviceEnv}_PORT_${port}_TCP_PORT`] = hostport;
-                preEnv[`${serviceEnv}_PORT_${port}_TCP_ADDR`] = hostname;
-            }
-        }
+    if (exposedPortOpt) {
+        const serviceEnv = makeServiceEnvName(exposedPortOpt.containerName);
+
+        const info = exposedPortOpt.loadBalancer;
+        const hostname = info.loadBalancer.dnsName;
+        const hostport = exposedPortOpt.containerPort.toString();
+        const hostproto = info.protocol;
+        const port = hostport;
```

Comment:
note this oddity.  we have hostport and port, but both are the same values.  We can see that from here: https://github.com/pulumi/pulumi-cloud/pull/662/files?w=1#diff-411860beb3c2bdcbd01894df3b3bd69aL726

Reply:
i've kept the variables to make teh diff easier t oread.